main.tf

```diff
@@ -19,7 +19,7 @@ resource "aws_backup_plan" "main" {
     for_each = var.rules
     content {
       rule_name           = lookup(rule.value, "name")
-      target_vault_name   = lookup(rule.value, "target_vault_name", null) == null ? var.vault_name : lookup(rule.value, "target_vault_name", "Default")
+      target_vault_name   = lookup(rule.value, "target_vault_name", "Default")
```

Comment:
The way I think this should be written is:
```
target_vault_name = var.vault_name != null ? aws_backup_vault.main[0]name : lookup(rule.value, "target_vault_name", "Default")
```

Please test with examples and if everything is working as expected then we can go for that.